### PR TITLE
Add doc strings to public entry points and their associated classes

### PR DESCRIPTION
**Pull request by shadeMe**
Also includes some minor renaming.

### CODE REVIEW

File: curated_transformers/models/bert/encoder.py [hunk 1]
```diff
@@ -41,10 +41,7 @@ def forward(
         Shapes:
             input_ids, token_type_ids - (batch, seq_len)
 
-        `attention_mask` indicates elements to attend to with `True` (and `False` otherwise)
-
-        Returns a tuple of consisting of a list of tensors from each Transformer
-        layer and the sum of the input and positional embeddings.
+        `attention_mask` indicates elements to attend to with `True` (and `False` otherwise).
```
Comment (danieldk) [inline curated_transformers/models/bert/encoder.py:44]: We should leave the true/false part out here, since we now have a dedicated class.

File: curated_transformers/models/roberta/encoder.py [hunk 2]
```diff
@@ -40,10 +40,7 @@ def forward(
         Shapes:
             input_ids, token_type_ids - (batch, seq_len)
 
-        `attn_mask` indicates elements to attend to with `1` (and `0` otherwise)
-
-        Returns a tuple of consisting of a list of tensors from each Transformer
-        layer and the sum of the input and positional embeddings.
+        `attn_mask` indicates elements to attend to with `1` (and `0` otherwise).
```
Comment (danieldk) [inline curated_transformers/models/roberta/encoder.py:43]: We should leave the 0/1 part out here, since we now have a dedicated class.

File: curated_transformers/models/transformer_model.py [hunk 3]
```diff
@@ -70,6 +70,52 @@ def build_albert_transformer_model_v1(
     mixed_precision: bool = False,
     grad_scaler_config: dict = SimpleFrozenDict(),
 ) -> TransformerModelT:
+    """Construct an ALBERT transformer model.
+
+    This model must be separately initialized using
```
Comment (danieldk) [inline curated_transformers/models/transformer_model.py:75]: I think this is worded too strongly. It is not necessary and in some cases desirable to not use a loader (e.g. train a model from scratch/distillation).

File: curated_transformers/pipe.py [hunk 4]
```diff
@@ -206,16 +235,19 @@ def predict(self, docs: Iterable[Doc]) -> TransformerModelOutput:
         return self.model.predict(docs)
 
     def set_annotations(
-        self, docs: Sequence[Doc], trf_output: TransformerModelOutput
+        self, docs: Sequence[Doc], predictions: TransformerModelOutput
```
Comment (danieldk) [inline curated_transformers/pipe.py:238]: I think I like the original name better. _predictions_ sound like things like class labels.

**Review by danieldk — COMMENTED** [review]
It's great to have all this documentation! Some minor comments.

File: curated_transformers/models/transformer_model.py (hunk 3 above)
Comment (shadeMe) [reply curated_transformers/models/transformer_model.py]: I've removed that sentence.

File: curated_transformers/pipe.py (hunk 4 above)
Comment (shadeMe) [reply curated_transformers/pipe.py]: Reverted.
